xrtpy/util/xrt_remove_lightleak.py

```diff
@@ -88,176 +200,46 @@ def xrt_remove_lightleak(in_map, kfact=1.0, leak_image=None, verbose=False):
-        where YYYY is the year, MM is the month number (zero-padded), DD is
-        the day of the month (zero-padded), HH is the hour and mm is the
-        minute (note T separator between date and time; may also include
-        seconds). This is the standard format for the value of date_obs in the
-        FITS headers of XRT images.
-
-    Returns:
-    --------
-    phase no. : `int`
-        The phase of the light leak as follows:
 
-    phase 0 : 23-Oct-2006 10:00
-    phase 1 :  9-May-2012 12:00
-    phase 2 : 14-Jun-2015 12:30
-    phase 3 : 27-May-2017 11:00
-    phase 4 : 29-May-2018 00:00
-    phase 5 :  8-Jun-2022 12:40
-    phase 6 :  5-May-2023  4:39
-    """
+    if leak_map is None:
+        leak_filename = _select_lightleak_file(
+            *in_map.measurement.split("-"), in_map.date
+        )
 
-    time_p1 = datetime.strptime("09-May-2012 12:00", "%d-%b-%Y %H:%M")
-    time_p2 = datetime.strptime("14-Jun-2015 12:30", "%d-%b-%Y %H:%M")
-    time_p3 = datetime.strptime("27-May-2017 11:00", "%d-%b-%Y %H:%M")
-    time_p4 = datetime.strptime("29-May-2018 00:00", "%d-%b-%Y %H:%M")
-    time_p5 = datetime.strptime("08-Jun-2022 12:40", "%d-%b-%Y %H:%M")
-    time_p6 = datetime.strptime("05-May-2023 04:39", "%d-%b-%Y %H:%M")
+        # NOTE: This function is being defined inline because the filename is only known once
+        # the date and filter wheel combination are known.
+        @manager.require(
+            "ll_file",
+            [
+                urljoin(mirror, f"hinode/xrt/idl/util/leak_fits/{leak_filename}")
+                for mirror in _SSW_MIRRORS
+            ],
+            LL_FILE_HASHES[leak_filename],
+        )
+        def get_ll_file():
+            return manager.get("ll_file")
 
-    intime = datetime.fromisoformat(date_obs)
-    if intime >= time_p6:
-        phase = 6
-    elif intime >= time_p5:
-        phase = 5
-    elif intime >= time_p4:
-        phase = 4
-    elif intime >= time_p3:
-        phase = 3
-    elif intime >= time_p2:
-        phase = 2
-    elif intime >= time_p1:
-        phase = 1
+        leak_map = Map(get_ll_file())
     else:
-        phase = 0
-
-    return phase
-
-
-def rebin(image, newshape):
-    """
-    Rebin an array to a new shape. Copied from:
-    https://scipy-cookbook.readthedocs.io/items/Rebinning.html
-    Should behave similarly to IDL rebin
-
-    Parameters:
-    -----------
-    image : 2D numpy array
-        image to be rebinned
+        # NOTE: This is to fill in the HISTORY in the resulting file
+        leak_filename = "unknown"
 
-    newshape : tuple
-        dimensions of rebinned image. Each dimension should be an integer
-        multiple of the corresponding dimension in the input image, though
-        they need not be the same multiple
+    if in_map.dimensions != leak_map.dimensions:
+        leak_map = leak_map.resample(u.Quantity(in_map.dimensions))
+        leak_map *= 0.25
```

Comment:
Should we add a comment here about why this is multiplied by 0.25?

Reply:
Yes, will do. Also need to restrict the condition to cases in which the in_map.dimensions are twice the leak_map.dimensions. 
I was also thinking that we should probably issue a warning if the image has the wrong dimensions generally. The use of the routine as written is supposed to be for composite images that are either 1024 x 1024 or 2048 x 2048. It won't work for images with different sizes.